src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -0,0 +1,779 @@
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+    });
+
+    it("refocuses the container within the trap when flag is set, if the wrapper has no tabindex", () => {
+      const { rerender } = render(
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+    });
+
+    it("refocuses the container if last element that had focus becomes disabled", () => {
+      const { rerender } = render(
+        <MockComponent autoFocus={false} triggerRefocusFlag={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input aria-label="input" type="text" id="disable-on-focus" />
+        </MockComponent>
+      );
+      focusElement(screen.getByLabelText("input"));
+      rerender(
+        <MockComponent autoFocus={false} triggerRefocusFlag>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" id="disable-on-focus" />
+        </MockComponent>
+      );
+      expect(screen.getByTestId(WRAPPER_ID)).toBeFocused();
+    });
+  });
+
+  describe("when autoFocus is false", () => {
+    it("should not focus the first focusable element by default", () => {
+      render(
+        <MockComponent autoFocus={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(document.querySelectorAll("body")[0]).toBeFocused();
+    });
+  });
+
+  describe("when isAnimationComplete is false", () => {
+    it("should not focus the first focusable element by default", () => {
+      render(
+        <MockComponent isAnimationComplete={false}>
+          <button type="button">{BUTTON_ONE}</button>
+          <input type="text" />
+        </MockComponent>
+      );
+      expect(document.querySelectorAll("body")[0]).toBeFocused();
+    });
+  });
+
+  describe("when a focusFirstElement is provided", () => {
+    beforeEach(() => {
+      render(
+        <MockComponent shouldFocusFirstElement>
+          <button type="button">{BUTTON_ONE}</button>
+          <button type="button">{BUTTON_TWO}</button>
+        </MockComponent>
+      );
+    });
+
+    it("should focus the element that ref passed to focusFirstElement and loop round when back tabbing", () => {
+      expect(screen.getByText(FIRST_ELEMENT)).toBeFocused();
+      shiftTabPress();
+      expect(screen.getByText(BUTTON_TWO)).toBeFocused();
+      shiftTabPress();
+      expect(screen.getByText(BUTTON_ONE)).toBeFocused();
+      shiftTabPress();
+      expect(screen.getByText(FIRST_ELEMENT)).toBeFocused();
+    });
```

Comment:
Could we split this into two tests here? One being `should focus one the first focusable child element` and `should loop back to first focusable element through back tabbing`

Reply:
I did consider this but ended up deciding that it was okay as one test as if I split I'd still be asserting that the first focusable element is focused in the looping one